samples/test/src/main.c

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2022 Golioth, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(test, LOG_LEVEL_DBG);
+#include <ztest.h>
+#include <net/golioth/system_client.h>
+
+static struct golioth_client *_client = GOLIOTH_SYSTEM_CLIENT_GET();
+
+ZTEST_SUITE(test_golioth, NULL, NULL, NULL, NULL, NULL);
+K_SEM_DEFINE(_connected_sem, 0, 1);
+
+static void golioth_on_connect(struct golioth_client *client)
+{
+	k_sem_give(&_connected_sem);
+}
+
+ZTEST(test_golioth, test_connect)
```

Comment:
Is there a reason to use ZTEST for that sample? Any advantages it gives in this context? So far I've seen ZTEST use only in unit/integration tests that were self-sufficient, i.e. without any external verification like the Python code in this sample.

Reply:
The truthful answer is - that's the way I did it in the ESP-IDF SDK, and it worked out okay (using Unity test framework which is similar to ZTEST).

But if I think a little deeper, I think there are two primary approaches:

1. ZTEST + FFF tests built into the firmware. Python does minimal serial IO to start the test and check the ZTEST output.
2. Shell commands in firmware, pytest is the main test controller. Code under test is activated via shell commands over serial from pytest.

I think there's merit to both approaches.

Option 1:
+ Has direct access to firmware functions and data. Better than option 2 where all control has to be routed through shell commands.
+ (minor) Does not require external python script to run tests, automatically run on boot
+ (minor) Same test frameworks used in all firmware tests (unit, integration, emulation, hardware)

Option 2:
+ pytest is a more powerful and flexible test framework

I think we should choose the option that seems like it will scale the best as we add more tests. I slightly prefer option 1, but I could be convinced that option 2 is better. 

What do you think?